src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -2719,15 +2921,16 @@ protected SocketAddress testableLocalSocketAddress() {
         return cnxn.sendThread.getClientCnxnSocket().getLocalSocketAddress();
     }
 
-    private static ClientCnxnSocket getClientCnxnSocket() throws IOException {
-        String clientCnxnSocketName = System
-                .getProperty(ZOOKEEPER_CLIENT_CNXN_SOCKET);
+    private ClientCnxnSocket getClientCnxnSocket() throws IOException {
```

Comment:
It is mostly a matter of separating concerns. It looks like we don't have to make the change of static -> instance, so we better not make the change if not needed. But, it also looks like the method was not accessing instance attributes previously and now it is. Let's keep it an instance method as you currently have it.